### PR TITLE
Introduce group tests (int/unit), add javadocs to server, add a script updating remote site

### PR DESCRIPTION
**Pull request by korotkevics**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -28,6 +28,7 @@
         <!--Testing-->
         <testngVersion>6.13.1</testngVersion>
         <mockitoVersion>2.13.0</mockitoVersion>
+        <failsafeVersion>2.20.1</failsafeVersion>
```
Comment (rivalez) [inline pom.xml]: wersje mozna wyniesc do properties

File: scripts/updateRemoteSite.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+# [How to use]
+# Execute it with no params ./updateRemoteSite.sh
+#
+# Have our remote site updated: https://korotkevics.github.io/ships/
+#
+# [Important]
+#
+# * It is dependent on clone.sh script
+# * It requires write access within a dir one level above the project root dir
+# * It takes a few minutes to execute
```
Comment (rivalez) [inline scripts/updateRemoteSite.sh]: few minutes hmm isn't it too much ?

File: scripts/updateRemoteSite.sh [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+# * It requires a user to provide his credentials at some stage (when performing git push)
+#
+
```
Comment (rivalez) [inline scripts/updateRemoteSite.sh]: set -e

File: pom.xml (hunk 1 above)
Comment (korotkevics) [reply pom.xml]: isn't it within properties?:)

File: server/src/main/java/com/epam/ships/server/Main.java [hunk 4]
```diff
@@ -5,9 +5,18 @@
 
 import java.io.IOException;
 
+/**
+ * * Main class of server application.
```
Comment (rivalez) [inline server/src/main/java/com/epam/ships/server/Main.java]: double "*" mark. Maybe consider renaming class Main.java to more something more meaningful.

File: scripts/updateRemoteSite.sh (hunk 2 above)
Comment (korotkevics) [reply scripts/updateRemoteSite.sh]: it is unavoidable here:( a github plugin which is not that useful seemed to run even longer

File: pom.xml (hunk 1 above)
Comment (rivalez) [reply pom.xml]: what do you mean by that? I was thinking about using one pom with versions only

Comment (korotkevics) [reply pom.xml]: ehm is there such a requirement? we have all versions in the root pom

Comment (rivalez) [reply pom.xml]: ahh ok my bad I was thinking about separate file for that but ok that is fine then.

File: scripts/updateRemoteSite.sh (hunk 3 above)
Comment (korotkevics) [reply scripts/updateRemoteSite.sh]: done:) thx

File: server/src/main/java/com/epam/ships/server/Main.java (hunk 4 above)
Comment (korotkevics) [reply server/src/main/java/com/epam/ships/server/Main.java]: @sabath91 please have a look

File: pom.xml (hunk 1 above)
Comment (korotkevics) [reply pom.xml]: thx anyways:)

File: server/src/main/java/com/epam/ships/server/Main.java (hunk 4 above)
Comment (sabath91) [reply server/src/main/java/com/epam/ships/server/Main.java]: Star has been removed. I leave this class name.

**Pull request merged**